File: src/main/java/com/tranhuudat/prms/service/ProjectService.java
package com.tranhuudat.prms.service;

import com.tranhuudat.prms.dto.BaseResponse;
import com.tranhuudat.prms.dto.project.ProjectDTO;
import com.tranhuudat.prms.dto.project.ProjectSearchRequest;

import java.util.UUID;

public interface ProjectService {

    BaseResponse create(ProjectDTO request);

    BaseResponse update(UUID id, ProjectDTO request);

    BaseResponse delete(UUID id);

    BaseResponse getById(UUID id);

    BaseResponse getPage(ProjectSearchRequest request);
}

File: prms-web/src/app/pages/project/services/project.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from '../../../core/services/base-service';
import { ProjectWritePayload } from '../models/project.model';
import { ProjectSearchRequest } from '../models/project-search.request';

@Injectable({
  providedIn: 'root',
})
export class ProjectService {
  private readonly apiUrl = '/api/v1/projects';

  constructor(private base: BaseService) {}

  getPage(request: ProjectSearchRequest) {
    return this.base.post(this.apiUrl + '/page', request);
  }

  create(request: ProjectWritePayload) {
    return this.base.post(this.apiUrl, request);
  }

  getById(id: string) {
    return this.base.get(`${this.apiUrl}/${id}`);
  }

  update(id: string, request: ProjectWritePayload) {
    return this.base.put(`${this.apiUrl}/${id}`, request);
  }

  delete(id: string) {
    return this.base.delete(`${this.apiUrl}/${id}`);
  }
}
